FungibleToken.ts

```diff
@@ -45,29 +47,30 @@ export class FungibleToken extends TokenContract implements FungibleTokenLike {
   async deploy(props: FungibleTokenDeployProps) {
     await super.deploy(props)
 
-    this.owner.set(props.owner)
+    this.admin.set(props.admin)
     this.supply.set(props.supply)
     this.circulating.set(UInt64.from(0))
 
     this.account.tokenSymbol.set(props.symbol)
     this.account.zkappUri.set(props.src)
   }
 
-  private ensureOwnerSignature() {
-    const owner = this.owner.getAndRequireEquals()
-    return AccountUpdate.createSigned(owner)
+  public getAdminContract(): FungibleTokenAdmin {
+    return (new FungibleTokenAdmin(this.admin.getAndRequireEquals()))
```

Comment:
would be cleaner if as return type this would have a minimal interface, not the precise `FungibleTokenAdmin` type. because the whole idea is that you should be able to change what this method returns, as long a you're returning a smart contract that has a `canAdmin()` method